### ISSUE TITLE
Many pin definitions seem incorrect in printer.cfg

### ISSUE THREAD

**Issue by Rygle** [LeeOtts/Ender3v2-Klipper-Configs#25]
Should be - according to your own 4.2.7 schematic and many other sources

[stepper_x]
step_pin: PC2
dir_pin: PB9

[stepper_y]
step_pin: PB8
dir_pin: PB7

[stepper_z]
step_pin: PB6
dir_pin: !PB5

**Comment by joennematti** [comment]
Thats right, I had to troubleshoot for quite some time until I switched the pin config around. It works now. Thanks for the info!

**Comment by maverickgu** [comment]
For the Creality 4.2.7 "Silent" Board, the following is the correct pinout for the X-axis:

step_pin: PB9

dir_pin: PC2

Why "PB9 / PC2" is the correct one:
On the 4.2.7 board, the STM32 chip is hard-wired to the TMC2225 (or TMC2208) drivers. Klipper needs to send the "pulse" to move on the step_pin and the "high/low" signal for direction on the dir_pin. If you swap these (using PC2 as the step pin), the motor will either vibrate loudly without moving or simply do nothing because the driver isn't receiving a valid clock signal on its Step input.

Refer to klipper github, the generic 4.2.7 points to the same mapping. What you had posted is for 4.2.2 board.

[klipper](https://github.com/bigtreetech/klipper/tree/master)/[config](https://github.com/bigtreetech/klipper/tree/master/config)
/generic-creality-v4.2.7.cfg

[stepper_x]
step_pin: PB9
dir_pin: PC2
enable_pin: !PC3
microsteps: 16
rotation_distance: 40
endstop_pin: ^PA5
position_endstop: 0
position_max: 235
homing_speed: 50

[stepper_y]
step_pin: PB7
dir_pin: PB8
enable_pin: !PC3
microsteps: 16
rotation_distance: 40
endstop_pin: ^PA6
position_endstop: 0
position_max: 235
homing_speed: 50
